cpp/src/community/leiden.cu

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <community/leiden.cuh>
+
+namespace cugraph {
+namespace detail {
+
+template <typename vertex_t, typename edge_t, typename weight_t>
+std::pair<int, weight_t> leiden(raft::handle_t const &handle,
+                                GraphCSRView<vertex_t, edge_t, weight_t> const &graph,
+                                vertex_t *leiden_parts,
+                                int max_level,
+                                weight_t resolution,
+                                cudaStream_t stream)
+{
+  CUGRAPH_EXPECTS(graph.edge_data != nullptr, "API error, leiden expects a weighted graph");
+  CUGRAPH_EXPECTS(leiden_parts != nullptr, "API error, leiden_parts is null");
+
+  Leiden<GraphCSRView<vertex_t, edge_t, weight_t>> runner(handle, graph, stream);
+
+  return runner.compute(leiden_parts, max_level, resolution);
```

Comment:
`Leiden` class should perhaps be a functor, with overloaded `operator()(leiden_parts, max_level, resolution)`. Rather than calling `runner.compute(leiden_parts, max_level, resolution)` the call would just be `runner(leiden_parts, max_level, resolution)`.

Passing functors is a natural expectation for STL/Thrust algorithms.

Reply:
Done in next commit